consensus/spos/bls/subroundBlock.go

```diff
@@ -371,6 +406,11 @@ func (sr *subroundBlock) createHeader() (data.HeaderHandler, error) {
 		return nil, err
 	}
 
+	if sr.EnableEpochsHandler().IsFlagEnabled(common.ConsensusPropagationChangesFlag) {
+		prevBlockProof := sr.Blockchain().GetCurrentHeaderProof()
```

Comment:
this is not complete: for the first block in the new epoch, right after transition, this proof would be empty, thus it won't advance

**it is already fixed in a further PR**